2006-04-04  Alexandre Julliard <julliard@winehq.org>

	* tools/winegcc/winegcc.c:
	winegcc: Don't use the library full name for .so libs.

Index: tools/winegcc/winegcc.c
===================================================================
RCS file: /home/wine/wine/tools/winegcc/winegcc.c,v
retrieving revision 1.50
retrieving revision 1.51
diff -u -p -r1.50 -r1.51
--- tools/winegcc/winegcc.c	16 Mar 2006 20:41:40 -0000	1.50
+++ tools/winegcc/winegcc.c	4 Apr 2006 21:16:14 -0000	1.51
@@ -401,8 +401,6 @@ static void add_library( strarray *lib_d
         }
         break;
     case file_so:
-        strarray_add(files, strmake("-s%s", fullname));
-        break;
     default:
         /* keep it anyway, the linker may know what to do with it */
         strarray_add(files, strmake("-l%s", library));
@@ -628,9 +626,9 @@ static void build(struct options* opts)
 	switch(files->base[j][1])
 	{
 	    case 'l':
+	    case 's':
 		strarray_add(link_args, strmake("-l%s", name));
 		break;
-	    case 's':
 	    case 'a':
 	    case 'o':
 		strarray_add(link_args, name);
